src/vm/comcallablewrapper.cpp

```diff
@@ -3969,27 +3945,26 @@ ComCallWrapperCache::~ComCallWrapperCache()
 // ComCallable wrapper manager
 // Create/Init method
 //-------------------------------------------------------------------
-ComCallWrapperCache *ComCallWrapperCache::Create(AppDomain *pDomain)
+ComCallWrapperCache *ComCallWrapperCache::Create(LoaderAllocator *pLoaderAllocator)
 {
     CONTRACT (ComCallWrapperCache*)
     {
         THROWS;
         GC_TRIGGERS;
         MODE_ANY;
         INJECT_FAULT(COMPlusThrowOM());
-        PRECONDITION(CheckPointer(pDomain));
```

Comment:
Isn't the precondition still valid? Seems like it is, unless this function is private and will never be called incorrectly.

Reply:
It is not valid since there is no pDomain argument anymore. However, it should be changed to the same check for pLoaderAllocator. I have had a state in the past where this function was still getting AppDomain and the cache was placed in both LoaderAllocator and AppDomain. So the AppDomain could have been NULL for the case when the wrapper was used from LoaderAllocator.
The current state is different and so we should check for the pLoaderAllocator here.